tests/Unit/Lti/Extractor/ReplaceResultSourceIdExtractorTest.php

```diff
@@ -42,22 +48,58 @@ public function testItThrowsInvalidLtiReplaceResultBodyExceptionOnInvalidXmlCont
     {
         $this->expectException(InvalidLtiReplaceResultBodyException::class);
 
-        $subject = new ReplaceResultSourceIdExtractor();
+        $subject = new ReplaceResultSourceIdExtractor(self::LTI_OUTCOME_XML_NAMESPACE);
 
         $subject->extractSourceId('invalid');
     }
 
+    public function testItThrowsInvalidLtiReplaceResultBodyExceptionOnInvalidXmlNamespace(): void
+    {
+        $this->expectException(InvalidLtiReplaceResultBodyException::class);
+
+        $subject = new ReplaceResultSourceIdExtractor(self::LTI_OUTCOME_XML_NAMESPACE);
+
+        /** @var string $xmlContent */
+        $xmlContent = file_get_contents(
+            dirname(__DIR__, 3) . DIRECTORY_SEPARATOR
+            . 'Resources/LtiOutcome/invalid_replace_result_body_wrong_namespace.xml'
+        );
+
+        $subject->extractSourceId($xmlContent);
+    }
+
     public function testItThrowsInvalidLtiReplaceResultBodyExceptionOnMissingId(): void
     {
         $this->expectException(InvalidLtiReplaceResultBodyException::class);
 
-        $subject = new ReplaceResultSourceIdExtractor();
+        $subject = new ReplaceResultSourceIdExtractor(self::LTI_OUTCOME_XML_NAMESPACE);
 
         /** @var string $xmlContent */
         $xmlContent = file_get_contents(
-            __DIR__ . '/../../../Resources/LtiOutcome/invalid_replace_result_body_missing_id.xml'
+            dirname(__DIR__, 3) . DIRECTORY_SEPARATOR
+            . 'Resources/LtiOutcome/invalid_replace_result_body_missing_id.xml'
         );
 
         $subject->extractSourceId($xmlContent);
     }
+
+    public function testItCanExtractSourceIdWithEnvironmentalValue(): void
```

Comment:
i know this is bit weird,  and i have received the feedback from @gabrielfs7 that we should not keep this , but i wanted to showcase that we are taking the value from the environment as this is being resolved by Symfony DI 
so please let me know if i should remove this

Reply:
You can remove it, I think we don't need it (we don't want to test the framework, only the application)

Reply:
done